### PR TITLE
[draft][browser] enable closure compiler for dotnet.js

### LINKED ISSUE #51524: [wasm] Enable closure compiler for dotnet.js
See https://github.com/dotnet/runtime/pull/51446, initial estimate of 15K compressed savings.

cc: @benaadams

### PR DESCRIPTION
**Pull request by pavelsavara**
Fixes https://github.com/dotnet/runtime/issues/51524

emcc `--closure 1`

Release size reduction of `dotnet.js`
from `338118` to `274571` = `63547`

7kb difference after brotli compression

### CODE REVIEW

**Comment by pavelsavara** [comment]
image

7kb after brotli compression

**Comment by pavelsavara** [comment]
With `compilation_level=SIMPLE_OPTIMIZATIONS` there is middle ground result. 
uncompressed `310860` and `70321` brotli

File: src/mono/wasm/runtime/es6/dotnet.es6.lib.js [hunk 1]
```diff
@@ -5,16 +5,16 @@
 "use strict";
 
 #if USE_PTHREADS
-const usePThreads = `true`;
+const usePThreads = true;
 const isPThread = `ENVIRONMENT_IS_PTHREAD`;
 #else
-const usePThreads = `false`;
+const usePThreads = false;
 const isPThread = `false`;
```
Comment (ilonatommy) [inline src/mono/wasm/runtime/es6/dotnet.es6.lib.js:12]: Why we are changing some strings to bools and some not?

Comment (pavelsavara) [reply src/mono/wasm/runtime/es6/dotnet.es6.lib.js:12]: because some are evaluated directly and some are appended in template.

**Comment by pavelsavara** [comment]
I should consider `SIMPLE_OPTIMIZATIONS` as default.

**Comment by eshadoel** [comment]
> Fixes #51524
> 
> emcc `--closure 1`
> 
> Release size reduction of `dotnet.js` from `338118` to `274571` = `63547`
> 
> 7kb difference after brotli compression

**Comment by pavelsavara** [comment]
Closing for now. We would have to ship Java as part of the wasm-tools workload.

**Pull request closed without merging**